app/controllers/photos_controller.rb

```diff
@@ -0,0 +1,35 @@
+class PhotosController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+  def index
+    @photos = Photo.all
+  end
+  def new
+    @photo = Photo.new
+  end
+  def create
+    current_user.photos.create(photos_params)
+    redirect_to '/photos'
+  end
+  def show
+    @photo = Photo.find(params[:id])
```

Comment:
This is repeated a few times. Can you think of a way of refactoring this? hint: private method.